job/src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -184,13 +184,11 @@ public WorkflowRun(WorkflowJob job, File dir) throws IOException {
             newExecution.addListener(new GraphL());
             completed = new AtomicBoolean();
             logsToCopy = new LinkedHashMap<String,Long>();
-            newExecution.start();
-            
-            // It's only okay to initialise the global once the new FlowExecution
-            // has successfully started.
             execution = newExecution;
+            newExecution.start();
```

Comment:
Should this not be on the line before so as to prevent `execution` from being initialized when `newExecution.start()` fails?